Source/Orts.Simulation/Simulation/Container.cs

```diff
@@ -531,6 +542,20 @@ public void Preload(string loadFilePath, int stackLocationIndex)
                 container.LoadFromContainerFile(loadFilePath, Simulator.BasePath + @"\trains\trainset\");
                 ContainerManager.LoadedContainers.Add(loadFilePath, container);
             }
+            switch (loadState)
+            {
+                case LoadState.Empty:
+                    container.MassKG = container.EmptyMassKG;
+                    break;
+                case LoadState.Loaded:
+                    container.MassKG = container.MaxMassWhenLoadedKG;
+                    break;
+                case LoadState.Random:
+                    var loadPercent = Simulator.Random.Next(101);
```

Comment:
Why don't you just perform a linear mapping?
```csharp
container.MassKG = container.EmptyMassKG + (container.MaxMassWhenLoadedKG-container.EmptyMassKG)*loadPercent/100f;
```

Reply:
Sorry, but I don't understand your point. I assume we are speaking about the LoadState.Random case.
With my code there is 30% of probability that the container is considered as empty, while there is a 70% of probability that the container weighs between 30% and 100% of the max allowable weight, because I assume that in general the weight of the freight is at least 30% of the weight of the max allowable weight. These are of course only assumptions, but I think they can be accepted.

Reply:
Ah, I see, thanks.